src/basecamp.h

```diff
@@ -68,7 +69,7 @@ class basecamp
         bool any_has_level( const std::string &type, int min_level ) const;
         bool can_expand() const;
         /// Returns the name of the building the current building @ref dir upgrades into, "null" if there isn't one
-        const std::string next_upgrade( const std::string &dir ) const;
+        const std::string next_upgrade( const std::string &dir, const int &offset ) const;
```

Comment:
```suggestion
        const std::string next_upgrade( const std::string &dir, const int offset = 1 ) const;
```
No need to pass an int as a reference.
Give the int a default value so you don't have to change all the other code.